src/html/parse-frontmatter.js

```diff
@@ -189,7 +189,7 @@ const findFrontmatter = (mdast, str) => {
           + 'and your horizontal rules have an empty line before AND after them.');
     }
 
-    const txt = str.slice(fst.offEnd, last.offStart);
+    const txt = str.slice(fst.offEnd, last === null ? undefined : last.offStart);
```

Comment:
```suggestion
    const txt = str.slice(fst.offEnd, !last ? undefined : last.offStart);
```

that may be a bit shorter. I'm not sure if `null === undefined`.